src/core/stripe/services/Stripe.ts

```diff
@@ -28,15 +28,22 @@ const createCheckoutSession: (params: StripeCheckoutParams ) => Promise<string>
         cancel_url: `${process.env.STRIPE_CANCEL_URL}`,
       }
 
-    // Add customer to session if exists
-    if (params.customerId !== null){
+    // Add customer to session if provided
+    if (params.customerId) {
         sessionParam.customer = params.customerId;
     } else {
-        // Add user email to session and create customer if not exists
-        sessionParam.customer_email = params.email;
-        sessionParam.customer_creation = 'always' as Stripe.Checkout.SessionCreateParams.CustomerCreation;
+        // Add customer email to session if provided
+        if (params.email) {
+            sessionParam.customer_email = params.email;
+        }
+        // Define customer creation
+        if (params.createCustomer) {
+            sessionParam.customer_creation = 'always' as Stripe.Checkout.SessionCreateParams.CustomerCreation;
+        }
+        else {
+            sessionParam.customer_creation = 'if_required' as Stripe.Checkout.SessionCreateParams.CustomerCreation;
```

Comment:
Im not completly sure but wouldnt 'if_required' be the default value for this field? In this case I think we can remove the 'else' clause.

Reply:
customer_creation is not being assigned automatically, it can't have a value if customer has one

Reply:
Based on the documentation if no value is sent in this parameter then no customer whatsoever is created. 'if_required' value will create a customer if the payment method is 'subscription'.
We will leave as it is as it covers most of the cases, is it pending to validate what happens if a subscription checkout is created without the customer_creation and customer parameters.
For more information see: https://stripe.com/docs/api/checkout/sessions/object#checkout_session_object-customer_creation